wasm.rkt

```diff
@@ -25,36 +26,46 @@
 
 (define-language WASM
   (c ::= real)
-  (e ::= binop relop (const c) (call i) trap debug-inst
+  (st ::= string)
+  (e ::= binop relop (const c) (call i) (call-indirect) trap debug-inst
      (get-local i) (set-local i) (local i) (param i)
      (block e ... end)
      (loop e ... end)
      (br j) (br-if j))
   (binop ::= add sub mul div)
   (relop ::= eq ne)
   (f ::= (func i e ...))
-  (m ::= (module f ...))
+  (mm ::= (ex ... memory c) )
+  (table-type ::= funcref anyfunc)
+  (tab ::= (table c table-type))
+  (elem-i ::= (elem v i i ...))
+  (ex ::= (export st i) )
+  ;(im ::= (import st x))
+  (m ::= (module tab elem-i ...  mm  f ...))
+  (v ::= (const c))
   (i ::= variable-not-otherwise-mentioned)
   (j ::= natural)
 
   ; Debugging statement, should never show up in a valid program
   (debug-inst ::= (debug any ...)))
 
 (define-extended-language WASM-eval WASM
+
   (op ::= binop relop)
   (init ::= (m e ...))
-  
+  (Σ-mem ::= mt-mem-store '())  
+  ;(elem-c ::=  v_offset)
   (s ::= (s-func s-table s-mem))
   (s-func ::= mt-s-func (i f s-func))
-  (s-table ::= mt-s-table)
-  (s-mem ::= mt-s-mem)
-  
+  (s-mem ::= mt-s-mem (c_memid v_index v Σ-mem s-mem))
+  (s-table ::= mt-s-table  (table-type c_size v_offset v_index i s-table))
```

Comment:
I do think we should change this to be something like:
`(s-table ::= mt-s-table  (table-type c_size v_offset table-elems)
(table-elems ::= (v_index i table-elems)`

what do you guys think?

Reply:
i have implemented what you suggested. Thanks for the suggestion, it was helpful